src/main/java/io/cdap/plugin/filecontent/FileContentsConfig.java

```diff
@@ -0,0 +1,190 @@
+import org.apache.hadoop.fs.Path;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import javax.annotation.Nullable;
+
+/**
+ * Config class that contains all properties required for running the unload command.
+ */
+public class FileContentsConfig extends PluginConfig {
+  public static final String SOURCE_FILE_PATH = "sourceFilePath";
+  public static final String FILE_REGEX = "fileRegex";
+  public static final String FILE_CONTENTS_REGEX = "fileContentsRegex";
+  public static final String FAIL_ON_EMPTY_FILE = "failOnEmptyFile";
+
+  @Name(SOURCE_FILE_PATH)
+  @Macro
+  @Description("The source location where the file or files live. You can use glob syntax here such as *.dat.")
+  private String sourceFilePath;
+
+  @Name(FILE_REGEX)
+  @Macro
+  @Nullable
+  @Description("A regular expression for filtering files such as .*\\.txt")
+  private String fileRegex;
+
+  @Name(FILE_CONTENTS_REGEX)
+  @Macro
+  @Nullable
+  @Description("A regular expression for checking the contents of the file.")
+  private String fileContentsRegex;
+
+  @Name(FAIL_ON_EMPTY_FILE)
+  @Macro
+  @Description("Set to true if this plugin should fail if the file is empty.")
+  private Boolean failOnEmptyFile;
+
+  public FileContentsConfig(String sourceFilePath, String fileRegex, String fileContentsRegex,
+                            Boolean failOnEmptyFile) {
+    this.sourceFilePath = sourceFilePath;
+    this.fileRegex = fileRegex;
+    this.fileContentsRegex = fileContentsRegex;
+    this.failOnEmptyFile = failOnEmptyFile;
+  }
+
+  private FileContentsConfig(Builder builder) {
+    sourceFilePath = builder.sourceFilePath;
+    fileRegex = builder.fileRegex;
+    fileContentsRegex = builder.fileContentsRegex;
+    failOnEmptyFile = builder.failOnEmptyFile;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static Builder builder(FileContentsConfig copy) {
+    return builder()
+      .setSourceFilePath(copy.sourceFilePath)
+      .setFileRegex(copy.fileRegex)
+      .setFileContentsRegex(copy.fileContentsRegex)
+      .setFailOnEmptyFile(copy.failOnEmptyFile);
+  }
+
+  public String getSourceFilePath() {
+    return sourceFilePath;
+  }
+
+  @Nullable
+  public String getFileRegex() {
+    return fileRegex;
+  }
+
+  @Nullable
+  public String getFileContentsRegex() {
+    return fileContentsRegex;
+  }
+
+  public Boolean getFailOnEmptyFile() {
+    return failOnEmptyFile;
+  }
+
+  /**
+   * Validates the config parameters required for unloading the data.
+   */
+  public void validate(FailureCollector failureCollector) {
+    try {
+      if (!containsMacro(FILE_REGEX) && !Strings.isNullOrEmpty(fileRegex)) {
+        Pattern.compile(fileRegex);
+      }
+    } catch (PatternSyntaxException e) {
+      failureCollector.addFailure(
+        "The regular expression pattern provided to match files is not a valid regular expression.", null)
```

Comment:
include the exception message in the error message, otherwise the user won't know why it is invalid.

Reply:
Done.